crates/jxl-render/src/filter/x86_64.rs

```diff
@@ -0,0 +1,136 @@
+use std::arch::x86_64::*;
+
+use jxl_grid::SimpleGrid;
+use jxl_grid::SimdVector;
+
+type Vector = __m128;
+
+#[inline]
+fn weight_sse2(scaled_distance: Vector, sigma: f32, step_multiplier: Vector) -> Vector {
+    let neg_inv_sigma = Vector::splat_f32(6.6 * (std::f32::consts::FRAC_1_SQRT_2 - 1.0) / sigma).mul(step_multiplier);
+    let result = scaled_distance.muladd(neg_inv_sigma, Vector::splat_f32(1.0));
+    unsafe { _mm_max_ps(result, Vector::zero()) }
+}
+
+macro_rules! define_epf_sse2 {
+    { $($v:vis fn $name:ident ($width:ident, $kernel_diff:expr, $dist_diff:expr $(,)?); )* } => {
+        $(
+            $v fn $name(
+                input: &[SimpleGrid<f32>; 3],
+                output: &mut [SimpleGrid<f32>; 3],
+                sigma_grid: &SimpleGrid<f32>,
+                channel_scale: [f32; 3],
+                border_sad_mul: f32,
+                step_multiplier: f32,
+            ) {
+                let width = input[0].width();
+                let $width = width as isize;
```

Comment:
Is it declaring a variable with ident `$width`, but it's used later as `width`?

Reply:
We have both `width` and `$width`; it's macro implementation details. We need the grid width at macro invocation site, but in order to do that, the caller should give it an ident (which is matched by `$width:ident`).